packages/itwinui-react/src/core/Stepper/Stepper.tsx

```diff
@@ -69,22 +108,38 @@ export const Stepper = React.forwardRef((props, ref) => {
   return (
     <Box className={'iui-stepper'} ref={ref} {...rest}>
       <ol>
-        {steps.map((s, index) => (
-          <StepperStep
-            key={index}
-            index={index}
-            title={type === 'long' ? '' : s.name}
-            currentStepNumber={boundedCurrentStep}
-            totalSteps={steps.length}
-            type={type}
-            onClick={onStepClick}
-            description={s.description}
-          />
-        ))}
+        {steps.map((s, index) => {
+          const thisStepProps = stepProps?.(index);
+          const thisTrackContentProps = trackContentProps?.(index);
+          const thisCircleProps = circleProps?.(index);
+          const thisNameProps = nameProps?.(index);
+          return (
+            <StepperStep
+              stepProps={thisStepProps}
+              trackContentProps={thisTrackContentProps}
+              circleProps={thisCircleProps}
+              nameProps={thisNameProps}
+              key={index}
+              index={index}
+              title={type === 'long' ? '' : s.name}
+              currentStepNumber={boundedCurrentStep}
+              totalSteps={steps.length}
+              type={type}
+              onClick={onStepClick}
+              description={s.description}
+            />
+          );
+        })}
       </ol>
       {type === 'long' && (
-        <Box className='iui-stepper-steps-label'>
-          <Box as='span' className='iui-stepper-steps-label-count'>
+        <Box className={cx('iui-stepper-steps-label', labelProps?.className)}>
+          <Box
+            as='span'
+            className={cx(
```

Comment:
The `labelCountProps` should be spread before the class name is modified

Reply:
Done